ts/features/itwallet/common/store/reducers/index.ts

```diff
@@ -56,9 +51,9 @@ const migrations: MigrationManifest = {
 
   // Added authLevel to preferences store and set it to "L2" if eid is present
   "2": (state: PersistedState): PersistedState => {
-    const { lifecycle, preferences } = state as PersistedItWalletState;
-    // If the lifecycle is valid that means we have an eid, set the authLevel to "L2"
-    if (lifecycle === ItwLifecycleState.ITW_LIFECYCLE_VALID) {
+    const { credentials, preferences } = state as PersistedItWalletState;
+    // If we have an eid, set the authLevel to "L2"
+    if (credentials.eid) {
```

Comment:
I am a little dubious about this control. When I implemented this migration I had had problems with this check and had switched to the check via lifecycle. Have you by any chance tried to do any in-depth migration tests? I remember that I had found a borderline case that caused problems

Reply:
Good catch! I forgot about this case! 
Lifecycle is removed from the current store definition, but this does not mean that previous stores does not have its data stored. Maybe I can try to mock the store type to let the migration happen without eslint complaining. Looking into it.

Reply:
I've fixed the migration. We should never use the typed state in migrations for this reason, we can't be sure that types never changes between migrations. 
Also I've added migrations tests. Let me know!